front/pages/w/[wId]/builder/data-sources/new.tsx

```diff
@@ -78,35 +83,51 @@ export default function DataSourceNew({
     });
     if (exists) {
       setDataSourceNameError("A DataSource with the same name already exists");
-      return false;
+      valid = false;
     } else if (dataSourceName.length == 0) {
+      valid = false;
       setDataSourceNameError("");
-      return false;
     } else if (dataSourceName.startsWith("managed-")) {
       setDataSourceNameError(
         "DataSource name cannot start with the prefix `managed-`"
       );
-      return false;
+      valid = false;
       // eslint-disable-next-line no-useless-escape
     } else if (!dataSourceName.match(/^[a-zA-Z0-9\._\-]+$/)) {
       setDataSourceNameError(
         "DataSource name must only contain letters, numbers, and the characters `._-`"
       );
-      return false;
+      valid = false;
     } else {
+      edited = true;
```

Comment:
NIT: initiate valid to false, and set it here to true instead of in all if??

Reply:
That does not scale to having other field being possibly invalid?